plugins/modules/firewall_alias.py

```diff
@@ -0,0 +1,346 @@
+                "macaddress",
+                "bgpasn",
+                "dynamicipv6host",
+                "opnvpngroup",
+                "internal",
+                "external",
+            ],
+            "required": True,
+        },
+        "content": {"type": "list", "elements": "str", "required": False},
+        "protocol": {
+            "type": "list",
+            "elements": "str",
+            "required": False,
+            "choices": ["IPv4", "IPv6", ""],
+        },
+        "statistics": {"type": "bool", "required": False, "default": False},
+        "description": {"type": "str", "required": False},
+        "refreshfrequency": {"type": "dict", "required": False},
+        "interface": {"type": "str", "required": False},
+        "state": {
+            "type": "str",
+            "required": False,
+            "default": "present",
+            "choices": ["present", "absent"],
+        },
+    }
+
+    module: AnsibleModule = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True,
+    )
+
+    # https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html
+    # https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#return-block
+    result = {
+        "changed": False,
+        "invocation": module.params,
+        "diff": None,
+    }
+
+    # make description ansible-managed
+    description: Optional[str] = module.params["description"]
+
+    if description and ANSIBLE_MANAGED not in description:
+        description = f"{ANSIBLE_MANAGED} - {description}"
+    else:
+        description = ANSIBLE_MANAGED
+
+    module.params["description"] = description
+
+    ansible_alias: FirewallAlias = FirewallAlias.from_ansible_module_params(
+        module.params
+    )
+
+    ansible_alias_state: str = module.params.get("state")
+
+    type_bgpasn_param: str = (
+        module.params.get("content") if module.params.get("type") == "bgpasn" else None
+    )
+    type_dynamicipv6host_param: str = (
+        module.params.get("content")
+        if module.params.get("type") == "dynamicipv6host"
+        else None
+    )
+    type_opnvpngroup_param: str = (
+        module.params.get("content")
+        if module.params.get("type") == "opnvpngroup"
+        else None
+    )
+
+    with FirewallAliasSet() as alias_set:
+        if type_opnvpngroup_param and alias_set.opnsense_version <= "23.1":
+            module.fail_json(
+                msg=f"Parameter is not supported in OPNsense {alias_set.opnsense_version}",
+                details=str(UnsupportedModuleSettingError),
+            )
+
+        if (
+            type_dynamicipv6host_param
+            or type_bgpasn_param
+            and alias_set.opnsense_version < "23.7"
+        ):
+            module.fail_json(
+                msg=f"Parameter is not supported in OPNsense {alias_set.opnsense_version}",
+                details=str(UnsupportedModuleSettingError),
+            )
+
```

Comment:
IMO These validations should be moved to the FirewallAliasSet:add_or_update functions.

Reply:
Hi @DonGiovanni83!
Thanks for the suggestion, fixed in f4833aa.